proposal.rst

```diff
@@ -237,7 +261,115 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+
+**April 23 – May 14 2018: Community Bonding Period:**
+
+Expected Deliverables: All working data downloaded, and testing started, with pre-eliminary testing examples and reporting, solving any possible bugs.
+
+	1. Writing tests.
+
+	2. Setting up tools required.
+
+	Also, I would not be able to contribute very actively in the last week due to my end semester exams from May 7th - May 14th.
+
+
+
+**Week 1: May 14 - May 21**
+
+Testing and benchmarking period.
+
+Expected Deliverables: Benchmarked reports for each of the INEX2009 and FIRE dataset. And a report on the running. 
+
+	1. Run benchmarking tests on the data-sets (INEX and FIRE).
+
+	2. Report the evaluation to the mentors. 
+
+	3. Make sure xapian-letor is stable by reporting and fixing bugs if any.
+
+**Week 2: May 14 - May 28**
+
+Testing and benchmarking period.
+
+Expected Deliverables: Stability of xapian-letor stress tested with various dataset examples. 
+
+	1. Writing examples to test proper execution. 
+
+	2. Make sure xapian-letor is stable by reporting and fixing bugs if any.
+
+	3. Cleaning up any previous work. 
+
+
+
+**Week 3: May 28 - June 2:**
+
+The next goal would be to adding a regression to combine multiple tests. 
+
+Expected deliverables: A base code for the regression problem to be written. 
+
+	1. Implement the feedforward backpropagation algorithm to combine various rankers by assignming random weights and then let them adjust according to the algorithm and learning rate. 
+
+**Week 4: May 28 - June 8:**
+
+Expected deliverables: Most of the regression implemented along with most working completed. 
+
+	1. Add the regression to xapian-letor.
+
+
+**Week 5:  June 8 - June 15:**
+
+Expected deliverables: All of the regression code ready. cleaned up, and mergable.
+
+	1. Complete and clean out the code.
+
+	2. Phase 1 evaluation reportable.
+
```

Comment:
Keep this as a buffer time, you might need that to catch up on whatever is pending. You don't need 1 week for evaluation. You are evaluated by the mentors on whatever goals you have completed. There will be at max 1 hour of discussion.